strax/storage/mongo.py

```diff
@@ -133,46 +136,81 @@ def __init__(self, key, metadata, col):
         # _save_chunk_metadata for the first chunk. Nevertheless we need an
         # object in case there e.g. is no chunk.
         basic_meta['run_start_time'] = datetime.now(py_utc)
+        # If available later update with this value:
+        self.run_start = None
+        # This info should be added to all of the associated documents
+        self.basic_md = basic_meta
+
+        # For the metadata copy this too:
+        meta_data = basic_meta.copy()
+        meta_data['metadata'] = self.md
 
-        # Save object id to write other data to
-        self.id = self.col.insert_one(basic_meta).inserted_id
+        # Save object_ids for fast querying and updates
+        self.id_md = self.col.insert_one(meta_data).inserted_id
+        # Also save all the chunks
+        self.ids_chunk = {}
 
-    def _save_chunk(self, data, chunk_info, executor):
+    def _save_chunk(self, data, chunk_info, executor=None):
         """see strax.Saver"""
         chunk_number = chunk_info['chunk_i']
+        chunk_name = f'chunk_{chunk_number}'
 
+        aggregate_data = []
         # Remove the numpy structures and parse the data. The dtype information
         # is saved with the metadata so don't worry
         for row in data:
             ins = {}
             for key in list(data.dtype.names):
                 ins[key] = row[key]
             ins = remove_np(ins)
-            self.col.update_one({'_id': self.id},
-                                {'$addToSet': {f'chunk_{chunk_number}':
-                                                   {"data": ins}}
-                                 })
+            aggregate_data.append(ins)
+
+        # Get the document to update, if none available start a new one for
+        # this chunk
+        chunk_id = self.ids_chunk.get(chunk_name, None)
+        if chunk_id is not None:
+            self.col.update_one({'_id': chunk_id},
+                                {'$addToSet': {f'chunk_name.data': aggregate_data}})
+        else:
+            # Start a new document, update it with the proper information
+            doc = self.basic_md.copy()
+            doc['write_time'] = datetime.now(py_utc)
+            doc[chunk_name] = {"data": aggregate_data}
```

Comment:
This line here is going to make life difficult for the analyst. You'll end up with chunks looking like this:
```
{chunk_0: {data: ...}},
{chunk_1: {data: ...}},
...
```
You can't easily loop over this (the field you want is different for each document), projections become more difficult (strictly inclusive, not exclusive), indexing chunk number is impossible, and probably some other things. It would make life much simpler if we just make a `data` field, rather than `chunk_%i.data`, and have a separate `chunk_i` field.

Reply:
You are right, I considered it did not make much of a difference as people will use st.get_array but that is a lousy excuse for making confusing documents